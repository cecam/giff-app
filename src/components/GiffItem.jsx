import PropTypes from 'prop-types'

const GiffItem = ({title, url}) => {
    return ( 
        <div className="card animate__animated animate__fadeIn">
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    );
}

GiffItem.proptypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
}
 
export default GiffItem;