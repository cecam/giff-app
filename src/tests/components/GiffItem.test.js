import React from 'react';
import { shallow } from 'enzyme'

import GiffItem from '../../components/GiffItem'

describe('Test on <GiffItem />', () => {
    const title = 'Test title'
    const url = 'http://localhost:3000'

    const wrapper = shallow( <GiffItem title={title} url={url} /> )
    
    test('should print the component', () => {
        expect( wrapper ).toMatchSnapshot()
    })

    test('Should have a paragraph with title', () => {
        const p = wrapper.find('p')

        expect( p.text().trim() ).toBe(title.trim())
    })

    test('should the img as the same time than a url and alt prop', () => {
        const img = wrapper.find('img')

        expect( img.prop('src') ).toBe(url)
        expect( img.prop('alt') ).toBe(title)
    });

    test('should have the animate__fadeIn class', () => {
        const div = wrapper.find('div')
        const animationClass = 'animate__fadeIn'
        
        const className = div.prop('className')

        expect( className.includes(animationClass) ).toBe(true)
    });
})