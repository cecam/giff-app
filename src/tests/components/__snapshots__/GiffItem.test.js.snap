// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GiffItem /> should print the component 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Test title"
    src="http://localhost:3000"
  />
  <p>
    Test title
  </p>
</div>
`;
